Spigot-API-Patches/0291-Added-PlayerInteractAngerPiglinEvent.patch

```diff
@@ -0,0 +1,109 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Mon, 4 Jan 2021 23:47:38 -0800
+Subject: [PATCH] Added PlayerInteractAngerPiglinEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/player/PlayerInteractAngerPiglinEvent.java b/src/main/java/io/papermc/paper/event/player/PlayerInteractAngerPiglinEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..28c886cb2af018ee5f5000917f94372e62fa22d1
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/player/PlayerInteractAngerPiglinEvent.java
+@@ -0,0 +1,97 @@
++package io.papermc.paper.event.player;
++
++import org.bukkit.Location;
++import org.bukkit.entity.Piglin;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.player.PlayerEvent;
++import org.bukkit.inventory.InventoryHolder;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++import java.util.List;
++
++/**
++ * Called when a player interacts with a chest-like block
++ * or destroys a block tagged as guarded by piglins.
++ */
++public class PlayerInteractAngerPiglinEvent extends PlayerEvent implements Cancellable {
```

Comment:
This name is really weird. `AngerPiglinByPlayerInteractionEvent`?

Reply:
Almost all events that extend `PlayerEvent` start with "Player". What about `PlayerAngerPiglinByInteractionEvent`?

Reply:
so I did that